### PR TITLE
[NU-1173] Table editor

### PR DESCRIPTION
**Pull request by JulianWielga**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/client/src/components/graph/node-modal/editors/expression/Table/CellMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    return (
+        <MenuItem
+            onClick={() => {
+                onClick(indexes);
+            }}
+        >
+            <ListItemIcon>
+                <DeleteForever fontSize="small" />
+            </ListItemIcon>
+            <ListItemText>Remove {indexes.length > 1 ? "columns" : "column"}</ListItemText>
```
Comment (Dzuming) [inline designer/client/src/components/graph/node-modal/editors/expression/Table/CellMenu.tsx:91]: Can we use i18next plural feature here?
```suggestion
          <ListItemText>
             {i18next.t("cellMenu.deleteColumns", {
                    defaultValue_one: "Remove {{count}} column",
                    defaultValue_other: "Remove {{count}} columns",
                    count: indexes.length,
                })}
              </ListItemText>
```

File: designer/client/src/components/graph/node-modal/editors/expression/Table/CellMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    return (
+        <MenuItem
+            onClick={() => {
+                onClick(indexes);
+            }}
+        >
+            <ListItemIcon>
+                <DeleteForever fontSize="small" />
+            </ListItemIcon>
+            <ListItemText>Remove {indexes.length > 1 ? "rows" : "row"}</ListItemText>
```
Comment (Dzuming) [inline designer/client/src/components/graph/node-modal/editors/expression/Table/CellMenu.tsx:52]: Can we use i18next plural feature here?

```suggestion
            <ListItemText>
                {i18next.t("cellMenu.deleteRow", {
                    defaultValue_one: "Remove {{count}} row",
                    defaultValue_other: "Remove {{count}} rows",
                    count: indexes.length,
                })}
            </ListItemText>
```

File: designer/client/src/fixedPortal.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { css } from "@emotion/css";
+import React from "react";
+
+export const FixedPortal = () => (
```
Comment (Dzuming) [inline designer/client/src/fixedPortal.tsx:4]: Maybe we should rename this component to say explicitly that, it's needed for the glide data editor? I had a problem with understanding what was going on here and why, but finally i just found out that the library requires this snippet... https://docs.grid.glideapps.com/api/dataeditor#html-css-prerequisites

File: designer/client/src/components/graph/node-modal/editors/expression/Table/tableState.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    const toExpression = useMemo(() => getStringifier(language), [language]);
+
+    const [state, dispatch] = useReducer(reducer, emptyValue, (defaultValue) => fromExpression(expression, defaultValue));
+
+    useEffect(() => {
+        setRawExpression((current) => {
+            if (current === expression) {
+                return current;
+            }
+            dispatch({
```
Comment (Dzuming) [inline designer/client/src/components/graph/node-modal/editors/expression/Table/tableState.ts:211]: This whole logic looks magical to me, i needed a few seconds to understand what's going on and still don't know when this dispatcher will potentially call.

File: designer/client/src/components/graph/node-modal/editors/expression/Table/CellMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        </MenuItem>
+    );
+}
+
+export function ResetColumnWidthMenuItem({
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply designer/client/src/components/graph/node-modal/editors/expression/Table/CellMenu.tsx:57]: cannot reproduce

[the remark this answers is not in the document]
Comment (Dzuming) [reply designer/client/src/components/graph/node-modal/editors/expression/Table/CellMenu.tsx:65]: I re-tested it and also cannot reproduce this issue. I will resolve this discussion.

**Pull request merged**